src/toplevel.c

```diff
@@ -339,99 +336,87 @@ static int jl_eval_expr_with_compiler_p(jl_value_t *e, int compileloops, jl_modu
     return 0;
 }
 
+static jl_value_t *root_module_func=NULL;
+
+jl_module_t *jl_lookup_root_module(jl_value_t *key)
+{
+    if (root_module_func == NULL && jl_base_module != NULL)
+        root_module_func = jl_get_global(jl_base_module, jl_symbol("root_module"));
+    jl_value_t *m = NULL;
+    if (root_module_func != NULL) {
+        jl_value_t *rmargs[2] = {root_module_func, key};
+        m = jl_apply(rmargs, 2);
```

Comment:
the system state will get severely corrupted if you call `jl_apply` from this context